src/Orleans.Membership.ServiceFabric/FabricMembershipOracle.cs

```diff
@@ -170,40 +152,32 @@ public SiloStatus GetApproximateSiloStatus(SiloAddress siloAddress)
             }
         }
 
-        /// <summary>
-        /// Determine if the current silo is dead.
-        /// </summary>
-        /// <returns>The silo so ask about.</returns>
+        /// <inheritdoc />
         public bool IsDeadSilo(SiloAddress address)
         {
             if (address.Equals(this.SiloAddress)) return false;
             var status = this.GetApproximateSiloStatus(address);
             return status == SiloStatus.Dead;
         }
 
-        /// <summary>
-        /// Determine if the current silo is valid for creating new activations on or for directory lookups.
-        /// </summary>
-        /// <returns>The silo so ask about.</returns>
+        /// <inheritdoc />
         public bool IsFunctionalDirectory(SiloAddress siloAddress)
         {
             if (siloAddress.Equals(this.SiloAddress)) return true;
 
             var status = this.GetApproximateSiloStatus(siloAddress);
-            return !status.IsTerminating();
+            return !status.IsUnavailable();
```

Comment:
Why not `!(status.IsTerminating() || status.IsUnavailable())`?

Reply:
`IsUnavailable` is defined as `siloStatus.IsTerminating() || siloStatus == SiloStatus.None;`, so the additional call to `IsTerminating()` would be redundant.